cpp/include/copc-lib/io/base_reader.hpp

```diff
@@ -0,0 +1,43 @@
+#include "copc-lib/las/vlr.hpp"
+
+#include <istream>
+#include <limits>
+#include <map>
+#include <string>
+
+#include "copc-lib/copc/copc_config.hpp"
+#include "copc-lib/hierarchy/key.hpp"
+#include "copc-lib/las/points.hpp"
+#include "copc-lib/las/vlr.hpp"
+
+namespace copc
+{
+class BaseReader
+{
+  public:
+    BaseReader(std::istream *in_stream) : in_stream_(in_stream) { InitReader(); }
+
+  protected:
+    BaseReader() = default;
+
+    las::LazConfig las_config_;
+    std::map<uint64_t, las::VlrHeader> vlrs_; // maps from absolute offsets to VLR entries
+
+    std::istream *in_stream_;
```

Comment:
```suggestion
    std::istream *in_stream_{};
```

Reply:
Is it better practice to use a unique_ptr instead?

Reply:
I don't think we need a pointer at all but removing that would be pretty substantial